awx/api/serializers.py

```diff
@@ -2201,6 +2199,64 @@ def create(self, validated_data):
         return return_data
 
 
+class BulkHostDeleteSerializer(serializers.Serializer):
+    def validate(self, attrs):
+        request = self.context.get('request', None)
+        # Extract the hosts ids from the querystring
+        d = request.query_params.get('pk_ids', '')
+        if d == '':
+            raise serializers.ValidationError({"Error": 'No hosts ids to delete'})
+        ids = [int(pk) for pk in d.split(',')]
+        if len(ids) > settings.BULK_HOST_MAX_DELETE:
+            raise serializers.ValidationError({"Error": 'Number of hosts exceeds system setting BULK_HOST_MAX_DELETE'})
+
+        # Getting list of all host objects, filtered by the list of the hosts to delete
+        attrs['in_inv'] = Host.objects.get_queryset().filter(pk__in=ids)
+        if len(attrs['in_inv']) == 0:
+            raise serializers.ValidationError({"Error": "There are no hosts to delete"})
+
+        if len(attrs['in_inv']) < len(ids):
+            inv_ids = [host.id for host in attrs['in_inv']]
+            raise serializers.ValidationError(
+                {
+                    "Error": "Not all hosts are in the inventory",
+                    "hosts": list(set(ids).difference(inv_ids)),
+                }
+            )
+
+        # Getting list of all inventories that hosts to delete can be in
+        hosts_inv = set([host.inventory for host in attrs['in_inv']])
+        for inv in hosts_inv:
+            if inv.kind != '':
+                raise serializers.ValidationError(_('Hosts can only be deleted from manual inventories (not smart or constructed types).'))
+            if request and not request.user.is_superuser:
+                if request.user not in inv.admin_role:
+                    raise serializers.ValidationError(_(f'Inventory with id {inv.id} not found or lack permissions to delete hosts from.'))
```

Comment:
Experiement if you can do single query that finds out if user has inventory admin on all the inventories
```
Inventory.objects.filter(id__in=hosts_inv, admin_role__contains(request.user)
```
I don't know if thats actually possible but something to that effect